### ISSUE TITLE
Работа  с директориями в Windows

### ISSUE THREAD

**Issue by konstantisu** [bookpauk/inpx-web#38]
1. Было бы неплохо добавить ремарку в Readme, что пути в json-конфиге нужно указывать с двумя слешами \\ - ладно я через час тупизны вспомнил про эти "особенности", но новички налетят сразу... :)
2. с путями с кириллицей в UTF-8 работает нормально (да, это моветон, но иногда нужно, если <обновляемая> раздача создана именно с таким путем, чтобы не возиться с перименованиями постоянно) И вот тут возникает странный баг? Если inpx лежит в том же каталоге, что и архивы - все в порядке. Если же в соседнем - получается ошибка  вида 
3. 2024-09-28 10:41:07.895 FATAL ERROR: EISDIR: illegal operation on a directory, read
4.  Образец куска конфига для известной раздачи прилагаю. Если руками переносить файл inpx в каталог с базой (Libdir)- и оставить дефолтным параметр  "inpx" - все в порядке. Если же указать (правильный) путь в параметре "inpx": - валится с ошибкой выше.
5. "libDir": "D:\\DOWNLOAD\\_Lib.rus.ec - Официальная\\lib.rus.ec",
"inpx": "D:\\DOWNLOAD\\_Lib.rus.ec - Официальная",
два слеша   \ \ автоматически "поправились" на один в этом комменте, ну это понятно.

**Comment by fredy314** [comment]
Возможно ли сделать чтобы в libDir можно было указать несколько директорий? у меня вся библиотека на одном диске не поместилась пришлось часть на другой диск перенести, d:\Flibusta+c:\Flibusta 
или можно как-=то линки на файлы сделать в виндовсе чтобы inpx-web мог их прочесть с другого диска?

**Comment by Weegley** [comment]
> 1. Было бы неплохо добавить ремарку в Readme, что пути в json-конфиге нужно указывать с двумя слешами \ - ладно я через час тупизны вспомнил про эти "особенности", но новички налетят сразу... :)
> 
>     2. с путями с кириллицей в UTF-8 работает нормально (да, это моветон, но иногда нужно, если <обновляемая> раздача создана именно с таким путем, чтобы не возиться с перименованиями постоянно) И вот тут возникает странный баг? Если inpx лежит в том же каталоге, что и архивы - все в порядке. Если же в соседнем - получается ошибка  вида
> 
>     3. 2024-09-28 10:41:07.895 FATAL ERROR: EISDIR: illegal operation on a directory, read
> 
>     4. Образец куска конфига для известной раздачи прилагаю. Если руками переносить файл inpx в каталог с базой (Libdir)- и оставить дефолтным параметр  "inpx" - все в порядке. Если же указать (правильный) путь в параметре "inpx": - валится с ошибкой выше.
> 
>     5. "libDir": "D:\DOWNLOAD\_Lib.rus.ec - Официальная\lib.rus.ec",
>        "inpx": "D:\DOWNLOAD\_Lib.rus.ec - Официальная",
>        два слеша   \ \ автоматически "поправились" на один в этом комменте, ну это понятно.

Попробуйте для начала сделать слеши обратными:
 ```
"libDir": "e:/fb2.Flibusta.Net/",  
"inpx": "e:/fb2.Flibusta.Net/flibusta_fb2_local.inpx", 
```

Попробовал повторить структуру как у вас:
```
"libDir": "e:/fb2.Flibusta.Net-Официальная/fb2.Flibusta.Net/",  
"inpx": "e:/fb2.Flibusta.Net-Официальная/flibusta_fb2_local.inpx", 
```

Всё работает
У Вас config.json в UTF же сохранён?

**Comment by Weegley** [comment]
> Возможно ли сделать чтобы в libDir можно было указать несколько директорий? у меня вся библиотека на одном диске не поместилась пришлось часть на другой диск перенести, d:\Flibusta+c:\Flibusta или можно как-=то линки на файлы сделать в виндовсе чтобы inpx-web мог их прочесть с другого диска?

Покопайте в сторону "[Соединения](https://learn.microsoft.com/ru-ru/sysinternals/downloads/junction)" или [MergeFS](https://github.com/harryrose/MergeFS)

**Comment by Weegley** [comment]
>  или можно как-=то линки на файлы сделать в виндовсе чтобы inpx-web мог их прочесть с другого диска?

Средствами windows, к сожалению - никак.
Разве только попробовать смонтировать папки как отдельные диски и эти диски потом объединить с помощью чего то вроде [MergeFS](https://github.com/harryrose/MergeFS)

**Comment by Weegley** [comment]
>  или можно как-=то линки на файлы сделать в виндовсе чтобы inpx-web мог их прочесть с другого диска?

Можно попробовать символьными ссылками через mklink:
```
mklink d:\main_folder\file1.zip e:\part1\file1.zip
mklink d:\main_folder\file2.zip f:\part2\file2.zip
```
Проверил сейчас - вынес inpx файл в сторонку, создал символьную ссылку - работает.
Принимающая папка (та, где будет ссылка) должна быть на NTFS
Получается - можно большую либу раскидать хоть на сотню частей.

1. Создаём [пустую] папку откуда inpx-web будет брать данные
2. Создаём в ней ссылки на нужные файлы (и даже папки)
3. В конфиге все пути на эту папку
4. PROFIT

**Comment by bookpauk** [comment]
Weegley, Вам надо памятник поставить за поддержку проекта, благодарю за помощь.

**Comment by bookpauk** [comment]
> Возможно ли сделать чтобы в libDir можно было указать несколько директорий? у меня вся библиотека на одном диске не поместилась пришлось часть на другой диск перенести, d:\Flibusta+c:\Flibusta или можно как-=то линки на файлы сделать в виндовсе чтобы inpx-web мог их прочесть с другого диска?

Так-то фича полезная, надо будет сделать поддержку разнесения библиотеки по папкам, она же все время растет. Вынесу в отдельное обращение.
